Code/Source/sv4gui/Modules/ProjectManagement/sv4gui_ProjectManager.h

```diff
@@ -70,6 +69,55 @@ class SV4GUIMODULEPROJECTMANAGEMENT_EXPORT sv4guiProjectManager
 
     static bool DuplicateDirRecursively(const QString &srcFilePath, const QString &tgtFilePath);
 
+    template <typename TDataFolder>
```

Comment:
@gmaher Why is this code being moved into a header?  I prefer that we keep major chunks of code by convention in .cxx files.

Reply:
The CreateDataFolder function is a template function. If plugins want to use it with their own datanode type they need to know where the function declaration is, so it can't be in the cxx file, has to be header. See e.g. https://stackoverflow.com/questions/1022623/c-shared-library-with-templates-undefined-symbols-error

For the current plugins all the data node folders were forward declared in the main simvascular branch. I dont think we want people editing the main simvascular branch whenever a new data type node is introduced.